src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralService.kt

```diff
@@ -239,10 +244,42 @@ class ReferralService(
     submitAdditionalRiskInformation(referral, user)
     communityAPIReferralService.send(referral)
 
-    val sentReferral = referralRepository.save(referral)
-    eventPublisher.referralSentEvent(sentReferral)
+    referralRepository.save(referral)
+    eventPublisher.referralSentEvent(referral)
     supplierAssessmentService.createSupplierAssessment(referral)
-    return sentReferral
+    // This is done to not break the UI.
+    setDraftReferralFields(draftReferral, referral)
+    return referral
+  }
+
+  private fun createReferral(draftReferral: DraftReferral): Referral {
+    return Referral(
```

Comment:
We discussed moving this into `Referral` entity class but decided against it because a business transition has better place here